### ISSUE TITLE
De novo plant genome

### ISSUE THREAD

**Issue by hidvegin** [FelixKrueger/TrimGalore#91]
I have got 150x Illumina 150 bp PE reads, 60x linked reads (10x Chromium). I checked the reads quality with FastQC. Here is the results:

Linked reads:

Library 1_1: https://unidebhu-my.sharepoint.com/:u:/g/personal/hidvegi-norbert_mailbox_unideb_hu/EQD8p1xFsD1DpDhadByJ93MBR0uLtEWfSL6CSgCAc5qdQQ?e=SzGNBv
Library 2: https://unidebhu-my.sharepoint.com/:u:/g/personal/hidvegi-norbert_mailbox_unideb_hu/EStXmkaZ385PqcNbzfq9h_MBljkgMJikuPdjPjNiIe4daQ?e=PnnHj9
Library 3: https://unidebhu-my.sharepoint.com/:u:/g/personal/hidvegi-norbert_mailbox_unideb_hu/EYHY9DpfzZJDphF-tw-uSucBxSPF48m8gMsyEbo0z76hbg?e=e8EAwq
Library 4: https://unidebhu-my.sharepoint.com/:u:/g/personal/hidvegi-norbert_mailbox_unideb_hu/EcDwKM98a-VAp04l7nr5cX4BLewLfTM5VXHiDwH0NAI2Ag?e=9zAavD

Illumina PE:

Library 1_1: https://unidebhu-my.sharepoint.com/:u:/g/personal/hidvegi-norbert_mailbox_unideb_hu/ETBOGWLmnjFGjYn8jdKh8j8Bi9e-AOLd9uWWjamB3c3pHg?e=ZT7AjS
Library 1_2: https://unidebhu-my.sharepoint.com/:u:/g/personal/hidvegi-norbert_mailbox_unideb_hu/Eepkc-Vi8TZDie8dFkSUducBMGOrLcebSmOIkI4r8WtNkw?e=QyFLps
Library 2_1: https://unidebhu-my.sharepoint.com/:u:/g/personal/hidvegi-norbert_mailbox_unideb_hu/EUI7aQrKzNFNoIk3-ZsoD2sBREwYuOJ8lieFWc9XX5iYEg?e=27anG3
Library 2_2: https://unidebhu-my.sharepoint.com/:u:/g/personal/hidvegi-norbert_mailbox_unideb_hu/EcXrp8WHRTpCmmPbIP9iJO4BHvWUfaSL3djgriIrcS41CA?e=QgPzNk
Library 3_1: https://unidebhu-my.sharepoint.com/:u:/g/personal/hidvegi-norbert_mailbox_unideb_hu/EQDcxuK_NPJPmXgn4n3rY14B2JnBZ3ezERQSU1JjtcJkng?e=HuqixI
Library 3_2: https://unidebhu-my.sharepoint.com/:u:/g/personal/hidvegi-norbert_mailbox_unideb_hu/EaRPP_kbDLlLmEs-TePvX_kBNSaa2fWe9FcND-N_Lh9d0Q?e=Yqd2gc
Library 4_1: https://unidebhu-my.sharepoint.com/:u:/g/personal/hidvegi-norbert_mailbox_unideb_hu/EXRU-jo6gshDmwrid_R9VLsB0i1i3LXZzcCYRBkATuEstQ?e=b2Hjvv
Library 4_2: https://unidebhu-my.sharepoint.com/:u:/g/personal/hidvegi-norbert_mailbox_unideb_hu/EZpcLPk_jupOouAqTS1CsHwBhgZZxYjZWDheg2bhMuXi2w?e=lo7key

I think the linked reads looks good but maybe Illumina adapter contaminated. Could I trim the Illumina adapters with TrimGalore from the linked reads?
The Illumina PE library is worse. At the Per base sequence content, the libraries have got a warning. How could I handle this errors with TrimGalore? Should I trim some base from the 5' and 3' ends?

**Comment by FelixKrueger** [comment]
Hi @hidvegin 

I just had a look at the FastQC profiles you linked. The linked libraries look excellent, there is a tiny little bit of adapter contamination coming up towards the 3' end of the reads. A standard Trim Galore run should take care of this, e.g.:

```
trim_galore LC001.fastq.gz
```

The paired-end libraries appear to have been trimmed (with Trim Galore?) already:
- there is absolutely no adapter contamination visible
- the base composition plot shows the very typical drop of A at the 3'-end (which is the result of stringently trimming off the Illumina adapter as it happens to start with `AGATC...`). In this case it is perfectly fine to get a warning of possibly fail from FastQC

In summary, I think the linked reads might benefit (very slightly) from running a default Trim Galore run, the PE libraries should be ready to go as they are.

Cheers, Felix

**Comment by hidvegin** [comment]
Hi @FelixKrueger 

Thank you for your answer. Yes, I used TrimGalore for the Illumina PE reads. I tried your suggestion with trim_galore on the linked reads. Here is the results with FastQC after the trimming.

Library 1: https://unidebhu-my.sharepoint.com/:u:/g/personal/hidvegi-norbert_mailbox_unideb_hu/Edx6df1co7pClL83RqCOr8YBrcHiRmow_C3Vz8TJsZbZmg?e=8wN1uP
Library 2: https://unidebhu-my.sharepoint.com/:u:/g/personal/hidvegi-norbert_mailbox_unideb_hu/ETJZuHzMEFBGrCClkO8nlzQBJnfM6OlnwPOZM7WHj9euZQ?e=SgdXuI
Library 3: https://unidebhu-my.sharepoint.com/:u:/g/personal/hidvegi-norbert_mailbox_unideb_hu/EbgUDAj1E0ZCo1A33iaK_UwBmCxX_1SBr2CMhtlBsNHjmA?e=TI3Ry1
Library 4: https://unidebhu-my.sharepoint.com/:u:/g/personal/hidvegi-norbert_mailbox_unideb_hu/EeOKLBOqM8JKmtZTrNFTttABZxJGF9bf8BAcM8GsFA_CTA?e=CBXiPX

I would like to use the Illumina PE reads for de novo assembly with abyss-pe and scaffolding the contigs with the linked reads. But at the first tests I get a little bit wrong results with this Illumina PE reads. Here is the results at k-mer 120 and 128 without the linked reads, only the Illumina PE reads:

k-mer 120:

```
n	n:500	L50	LG50	NG50	min	N75	N50	N25	E-size	max	sum	name
11.4e6	1602860	359966	1602860	500	500	789	1398	2798	2176	29243	1.916e9	lculinaris-unitigs.fa
10.01e6	1284682	211374	1284682	500	500	901	2497	4869	3548	42918	1.972e9	lculinaris-contigs.fa
9964249	1247305	184732	1247305	500	500	900	2769	5595	4033	42918	1.971e9	lculinaris-scaffolds.fa
```

k-mer 128:

```
n	n:500	L50	LG50	NG50	min	N75	N50	N25	E-size	max	sum	name
9959811	1723659	413852	1723659	500	500	774	1308	2471	1950	28567	1.98e9	lculinaris-unitigs.fa
8587960	1364518	230144	1302252	515	500	876	2370	4616	3364	39903	2.033e9	lculinaris-contigs.fa
8533308	1319540	197069	1257274	515	500	876	2651	5426	3911	48103	2.032e9	lculinaris-scaffolds.fa
```
The N50 values is very low at k-mer 120 and 128 also. I think that maybe cause the Per base sequence content, but now you mentioned that this is not a problem and this reads are ready for de novo assembly.

**Comment by FelixKrueger** [comment]
From a trimming point of view, the samples look good:

- read quality is fine
- no adapter contamination left
- the 'blip' in the read composition at the 3' end is a an effect of the trimming, and should not inhibit the assembly

I am however not exactly sure why there are some irregularities at the 5' end of your reads, and whether or not this could be problematic or not. I guess this depends on what the PE reads actually are, and whether or not you would expect any irregularities at the start of reads. I suppose you could fairly quickly test this by using `--clip_r1 10 --clip_r2 10`, and then seeing if the assembly stats improve? But yea I think this is ultimately linked to your experiment, and not really something I am likely to be able to to help you with...

**Comment by hidvegin** [comment]
Should I trim all of the libraries (linked reads and Illumina PE) with `--clip_r1 10 --clip_r2 10`?

**Comment by FelixKrueger** [comment]
This would really just be a test to see if the sequence at the 5' end is inhibitory in some way or not. I have no experience in the assembly field I am afraid, so you should probably get tips from someone who knows what they are doing :)

**Comment by hidvegin** [comment]
Thank you I will try it.
